cbits/cbits.c

```diff
@@ -82,6 +87,23 @@ _hs_text_decode_latin1(uint16_t *dest, const uint8_t *src,
   while (p != srcend && (uintptr_t)p & 0x3)
     *dest++ = *p++;
 
+#if defined(__x86_64__)
+  /* All the intrinsics used here are from SSE2,
+   * so every x86_64 CPU supports them.
+   */
+  const __m128i zeros = _mm_set1_epi32(0);
+  while (p < srcend - 7) {
+    /* Load 8 bytes of ASCII data */
+    const __m128i ascii = _mm_cvtsi64_si128(*((const uint64_t *)p));
+    /* Interleave with zeros */
+    const __m128i utf16 = _mm_unpacklo_epi8(ascii, zeros);
```

Comment:
Naive question: how do we tell the arguments should be this way and not the other way around?

Reply:
The other way around would mean that for input bytes "xyzw" the output is not "x0y0z0w0" but "0x0y0z0w". 

The test suite can catch this (note that 'a' is 97 and 24832 is 97 * 256):

```
      t_latin1:                    FAIL
        *** Failed! Falsified (after 13 tests and 9 shrinks):
        "aaaaaaaa"
        "\24832\24832\24832\24832\24832\24832\24832\24832" /= "aaaaaaaa"
        Use --quickcheck-replay=816513 to reproduce.
```

I'm pretty bad at mental simulation of little-endian representation, so I usually just check which order of arguments results in tests passing.